go/graph/search.go

```diff
@@ -52,3 +52,62 @@ func (g *Graph) DijkstraSearch(start Node) []Path {
 	}
 	return paths
 }
+
+// Context for an AllPathSearch
+type allPathSearch struct {
+	// Graph
+	Graph *Graph
+	// Current path
+	Path []Edge
+	// Set of node values on the path
+	NodeValues map[*interface{}]bool
+	// Final paths
+	Result []Path
+}
+
+// AllPathSearch returns all the paths from the start node to another
+// node in the graph. The graph can be cyclic but cyclic paths will never be included.
```

Comment:
Good comment :+1: - can the formatting be fixed? (`gwap` in normal mode in vim, if you use it)